general/src/datascrubbing/convert.rs

```diff
@@ -0,0 +1,2809 @@
+
+        insta::assert_snapshot_matches!(data.to_json_pretty().unwrap(), @r###"
+       ⋮{
+       ⋮  "stacktrace": {
+       ⋮    "frames": [
+       ⋮      {
+       ⋮        "vars": {
+       ⋮          "a_password_here": null,
+       ⋮          "apiKey": null,
+       ⋮          "api_key": null,
+       ⋮          "foo": "bar",
+       ⋮          "password": null,
+       ⋮          "the_secret": null
+       ⋮        }
+       ⋮      }
+       ⋮    ]
+       ⋮  },
+       ⋮  "_meta": {
+       ⋮    "stacktrace": {
+       ⋮      "frames": {
+       ⋮        "0": {
+       ⋮          "vars": {
+       ⋮            "a_password_here": {
+       ⋮              "": {
+       ⋮                "rem": [
+       ⋮                  [
+       ⋮                    "@password",
+       ⋮                    "x"
+       ⋮                  ]
+       ⋮                ]
+       ⋮              }
+       ⋮            },
+       ⋮            "apiKey": {
+       ⋮              "": {
+       ⋮                "rem": [
+       ⋮                  [
+       ⋮                    "@password",
+       ⋮                    "x"
+       ⋮                  ]
+       ⋮                ]
+       ⋮              }
+       ⋮            },
+       ⋮            "api_key": {
+       ⋮              "": {
+       ⋮                "rem": [
+       ⋮                  [
+       ⋮                    "@password",
+       ⋮                    "x"
+       ⋮                  ]
+       ⋮                ]
+       ⋮              }
+       ⋮            },
+       ⋮            "password": {
+       ⋮              "": {
+       ⋮                "rem": [
+       ⋮                  [
+       ⋮                    "@password",
+       ⋮                    "x"
+       ⋮                  ]
+       ⋮                ]
+       ⋮              }
+       ⋮            },
+       ⋮            "the_secret": {
+       ⋮              "": {
+       ⋮                "rem": [
+       ⋮                  [
+       ⋮                    "@password",
+       ⋮                    "x"
+       ⋮                  ]
+       ⋮                ]
+       ⋮              }
+       ⋮            }
+       ⋮          }
+       ⋮        }
+       ⋮      }
+       ⋮    }
+       ⋮  }
+       ⋮}
+        "###);
+    }
+
+    #[test]
+    fn test_http() {
+        let mut data = Event::from_value(
+            serde_json::json!({
+                "request": {
+                    "data": SENSITIVE_VARS.clone(),
+                    "env": SENSITIVE_VARS.clone(),
+                    "headers": SENSITIVE_VARS.clone(),
+                    "cookies": SENSITIVE_VARS.clone()
+                }
+            })
+            .into(),
+        );
+
+        let pii_config = get_default_pii_config();
+        let mut pii_processor = PiiProcessor::new(&pii_config);
+
+        process_value(&mut data, &mut pii_processor, ProcessingState::root());
+
+        insta::assert_snapshot_matches!(data.to_json_pretty().unwrap(), @r###"
+       ⋮{
```

Comment:
Putting this up for discussion: Since these snapshots are pretty large, can we not inline them?
To me, they are not consumable in this format either, and at least we'd declutter the file.

Reply:
I think it's good that the inline snapshots make large snapshots painful, because large snapshots are a mistake. I would actually opt for preferring inline snapshots for that reason alone.

I'll see what can be done about this. I believe we could snapshot single event keys and use more compact json serialization.

Reply:
I moved the largest snapshots out of sight... my opinion was colored by our experience with symbolicator snapshots, but it's actually fine.